cartographer/mapping/internal/2d/normal_estimation_2d_test.cc

```diff
@@ -0,0 +1,138 @@
+      "sample_radius = 10.0, "
+      "}");
+  const proto::NormalEstimationOptions2D options =
+      CreateNormalEstimationOptions2D(parameter_dictionary.get());
+  sensor::RangeData range_data;
+  range_data.returns.emplace_back(-1.f, 1.f, 0.f);
+  range_data.returns.emplace_back(0.f, 1.f, 0.f);
+  range_data.returns.emplace_back(1.f, 1.f, 0.f);
+  range_data.origin.x() = 0.f;
+  range_data.origin.y() = 0.f;
+  std::vector<float> normals;
+  EstimateNormals(range_data, &normals, options);
+  for (const float normal : normals) {
+    EXPECT_NEAR(normal, -M_PI_2, 1e-4);
+  }
+
+  normals.clear();
+  range_data.returns.clear();
+  range_data.returns.emplace_back(1.f, 1.f, 0.f);
+  range_data.returns.emplace_back(1.f, 0.f, 0.f);
+  range_data.returns.emplace_back(1.f, -1.f, 0.f);
+  EstimateNormals(range_data, &normals, options);
+  for (const float normal : normals) {
+    EXPECT_NEAR(std::abs(normal), M_PI, 1e-4);
+  }
+
+  normals.clear();
+  range_data.returns.clear();
+  range_data.returns.emplace_back(1.f, -1.f, 0.f);
+  range_data.returns.emplace_back(0.f, -1.f, 0.f);
+  range_data.returns.emplace_back(-1.f, -1.f, 0.f);
+  EstimateNormals(range_data, &normals, options);
+  for (const float normal : normals) {
+    EXPECT_NEAR(normal, M_PI_2, 1e-4);
+  }
+
+  normals.clear();
+  range_data.returns.clear();
+  range_data.returns.emplace_back(-1.f, -1.f, 0.f);
+  range_data.returns.emplace_back(-1.f, 0.f, 0.f);
+  range_data.returns.emplace_back(-1.f, 1.f, 0.f);
+  EstimateNormals(range_data, &normals, options);
+  for (const float normal : normals) {
+    EXPECT_NEAR(normal, 0, 1e-4);
+  }
+}
+
+TEST(NormalEstimation2DTest, CircularGeometry) {
+  const auto parameter_dictionary = common::MakeDictionary(
+      "return { "
+      "num_normal_samples = 2, "
+      "sample_radius = 10.0, "
+      "}");
+  const proto::NormalEstimationOptions2D options =
+      CreateNormalEstimationOptions2D(parameter_dictionary.get());
+  sensor::RangeData range_data;
+  const size_t num_angles = 100;
+  for (size_t angle_idx = 0; angle_idx < num_angles; ++angle_idx) {
+    const double angle = static_cast<double>(angle_idx) /
+                             static_cast<double>(num_angles) * 2. * M_PI -
+                         M_PI;
+    range_data.returns.emplace_back(std::cos(angle), std::sin(angle), 0.f);
+  }
+  range_data.origin.x() = 0.f;
+  range_data.origin.y() = 0.f;
+  std::vector<float> normals;
+  EstimateNormals(range_data, &normals, options);
+  for (size_t angle_idx = 0; angle_idx < num_angles; ++angle_idx) {
+    const double angle = static_cast<double>(angle_idx) /
+                         static_cast<double>(num_angles) * 2. * M_PI;
+    EXPECT_NEAR(common::NormalizeAngleDifference(normals[angle_idx] - angle),
+                0.0, 5e-2);
+  }
+}
+
+TEST(NormalEstimation2DTest, CircularGeometrySingleSample) {
+  const auto parameter_dictionary = common::MakeDictionary(
+      "return { "
+      "num_normal_samples = 1, "
```

Comment:
Since this test seems to differ only in one character, you could take a look if the twot can easier be written as parametric ones https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#how-to-write-value-parameterized-tests 
(I don't have a strong opinion here)

Reply:
Yes, I strongly agree. It makes the test more readable and removes redundancy.